src/mod/endpoints/mod_verto/mod_verto.c

```diff
@@ -1764,41 +1764,52 @@ static void http_run(jsock_t *jsock)
 		goto done;
 	}
 
-	if (!strncmp(request->method, "POST", 4) && request->content_length && request->content_type &&
-		!strncmp(request->content_type, "application/x-www-form-urlencoded", 33)) {
-
-		char *buffer = NULL;
-		switch_ssize_t len = 0, bytes = 0;
-
-		if (request->content_length && request->content_length > 10 * 1024 * 1024 - 1) {
-			char *data = "HTTP/1.1 413 Request Entity Too Large\r\n"
-				"Content-Length: 0\r\n\r\n";
-			kws_raw_write(jsock->ws, data, strlen(data));
-			request->keepalive = 0;
-			goto done;
-		}
+	if (!strncmp(request->method, "POST", 4) && request->content_length && request->content_type) {
+		switch_bool_t content_type_urlencoded = SWITCH_FALSE;
+		switch_bool_t content_type_json = SWITCH_FALSE;
+
+		content_type_urlencoded = !strncmp(request->content_type, "application/x-www-form-urlencoded", 33);
+		content_type_json = !strncmp(request->content_type, "application/json", 16);
+
+		if (content_type_urlencoded || content_type_json) {
+			char *buffer = NULL;
+			switch_ssize_t len = 0, bytes = 0;
+
+			if (request->content_length && request->content_length > 10 * 1024 * 1024 - 1) {
+				char *data = "HTTP/1.1 413 Request Entity Too Large\r\n"
+					"Content-Length: 0\r\n\r\n";
+				kws_raw_write(jsock->ws, data, strlen(data));
+				request->keepalive = 0;
+				goto done;
+			}
 
-		if (!(buffer = malloc(2 * 1024 * 1024))) {
-			goto request_err;
-		}
+			if (!(buffer = malloc(2 * 1024 * 1024))) {
+				goto request_err;
+			}
 
-		while(bytes < (switch_ssize_t)request->content_length) {
-			len = request->content_length - bytes;
+			while(bytes < (switch_ssize_t)request->content_length) {
+				len = request->content_length - bytes;
 
 #define WS_BLOCK 1
 
-			if ((len = kws_raw_read(jsock->ws, buffer + bytes, len, WS_BLOCK)) < 0) {
-				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "Read error %" SWITCH_SSIZE_T_FMT"\n", len);
-				goto done;
+				if ((len = kws_raw_read(jsock->ws, buffer + bytes, len, WS_BLOCK)) < 0) {
+					switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "Read error %" SWITCH_SSIZE_T_FMT"\n", len);
+					goto done;
```

Comment:
This is leaking the `buffer`

Reply:
Changed the buffer size to hold the message body content without overflow.

Reply:
Is memory freed in this case?

Reply:
No it wasn't, good catch. I fixed it.
Sorry initially I was hesitant to do improvement and cleanup in the original code assuming it will not be appreciated. But I will keep that in my mind next time.